Lib/idlelib/editor.py

```diff
@@ -1368,14 +1365,23 @@ def smart_backspace_event(self, event):
         have = len(chars.expandtabs(tabwidth))
         assert have > 0
         want = ((have - 1) // self.indentwidth) * self.indentwidth
+
         # Debug prompt is multilined....
         ncharsdeleted = 0
-        while True:
-            chars = chars[:-1]
-            ncharsdeleted = ncharsdeleted + 1
-            have = len(chars.expandtabs(tabwidth))
-            if have <= want or chars[-1] not in " \t":
+        have = len(chars.expandtabs(tabwidth))
+        for i in range(len(chars) - 1, -1, -1):
+            # ``Delete'' chars[i], and subtract count
+            # (since redoing expandtabs is O(n))
+            ncharsdeleted += 1
+            if chars[i] == '\t':
+                have -= tabwidth
+            else:
+                have -= 1
+            if have <= want or chars[i-1] not in " \t":
                 break
+        # Perform the actual removal
+        chars = chars[:len(chars) - ncharsdeleted]
+
```

Comment:
```suggestion
        chars = chars[:len(chars) - ncharsdeleted]
```

let's drop obvious comments